jetty-core/jetty-http2/jetty-http2-hpack/src/main/java/org/eclipse/jetty/http2/hpack/HpackDecoder.java

```diff
@@ -39,6 +39,7 @@
 public class HpackDecoder
 {
     private static final Logger LOG = LoggerFactory.getLogger(HpackDecoder.class);
+    private static final HttpField CONTENT_LENGTH_0 = new PreEncodedHttpField(HttpHeader.CONTENT_LENGTH, "content-length", "0");
```

Comment:
Use `HttpFields.CONTENT_LENGTH_0`

Reply:
Ah you are not using it because you want it lowercase!  A comment or a better name please so somebody does not revert it back:
```suggestion
    private static final HttpField CONTENT_LENGTH_LOWER_CASE_0 = new PreEncodedHttpField(HttpHeader.CONTENT_LENGTH, "content-length", "0");
```

Reply:
Hmmm explain exactly why this is required?  The pre-encoder is already using lowercase name.  Confused?

Reply:
I replaced with HttpFields.CONTENT_LENGTH_0 and no tests failed?

Reply:
@gregw I erroneously committed the check on the validity of the header using `getLowerCaseName()`, that's why reverting no test failed.

The problem is that the decoder had a special case for `HttpHeader.CONTENT_LENGTH`, where even when it was decoding the lower case name, it was then returning the `HttpFields.CONTENT_LENGTH_0` constant, which was camel-case, and was then cached.
On a subsequent request, we returned the cached field, which had the camel-case name, so a check on the name would fail.

Reply:
OK - so a comment and/or better name is needed